Assets/00_Content/Scripts/Interactables/Beers/BeerTap.cs

```diff
@@ -1,11 +1,35 @@
+using Extensions;
+using System.Collections;
 using UnityEngine;
+using UnityEngine.InputSystem;
+using UnityEngine.InputSystem.Controls;
+using UnityEngine.UI;
 
 namespace Interactables.Beers {
 
 	public class BeerTap : Interactable {
 
+		[Header("Settings")]
+		[Range(0.1f, 2f)]
+		[SerializeField] private float pourTimeMultiplier = 1f;
+
+		[Header("GameObjects")]
+		[SerializeField] private InputActionAsset InputActions;
 		[SerializeField] private GameObject beerPrefab;
 		[SerializeField] private Transform beerSpawnpoint;
+		[SerializeField] private Image progressBarImage;
+		[SerializeField] private GameObject progressBar;
+
+		private ItemSlot itemSlot;
+		private float pouringProgress = 0;
+		InputAction inputActionInteract;
+		ButtonControl buttonInteract;
+
+		private void Start() {
+			itemSlot = GetComponentInChildren<ItemSlot>();
+			inputActionInteract = InputActions.FindAction("Interact");
+			buttonInteract = (ButtonControl)inputActionInteract.controls[0];
```

Comment:
This button control represents the first control of the "Interact" action eg. the 'C' key. This means that gamepads are ignored.